test/NuGet.Core.Tests/NuGet.Packaging.Test/SigningTests/RepositoryCountersignatureTests.cs

```diff
@@ -0,0 +1,400 @@
+
+                // Generate valid countersignature
+                test.CounterCmsSigner.SignedAttributes.Add(
+                    AttributeUtility.CreateCommitmentTypeIndication(SignatureType.Repository));
+                test.CounterCmsSigner.SignedAttributes.Add(
+                     AttributeUtility.CreateNuGetV3ServiceIndexUrl(new Uri("https://api.nuget.org/v3/index.json")));
+                test.CounterCmsSigner.SignedAttributes.Add(
+                    new Pkcs9SigningTime());
+                test.CounterCmsSigner.SignedAttributes.Add(
+                    AttributeUtility.CreateSigningCertificateV2(test.Certificate, HashAlgorithmName.SHA256));
+
+                // Create counter signature
+                test.PrimarySignedCms.SignerInfos[0].ComputeCounterSignature(test.CounterCmsSigner);
+
+                // Load primary signature
+                var primarySignature = PrimarySignature.Load(test.PrimarySignedCms.Encode());
+
+                // Validate countersignature
+                var countersignature = RepositoryCountersignature.GetRepositoryCountersignature(primarySignature);
+
+                Assert.Equal(SignatureType.Repository, countersignature.Type);
+            }
+        }
+
+        [Fact]
+        public void GetRepositoryCounterSignature_WithValidRepositoryCountersignature_ReturnsRepositoryCountersignature()
+        {
+            using (var test = new LoadTest(_fixture))
+            {
+                test.CreateValidAuthorPrimarySignature();
+
+                // Generate valid countersignature
+                test.CounterCmsSigner.SignedAttributes.Add(
+                    AttributeUtility.CreateCommitmentTypeIndication(SignatureType.Repository));
+                test.CounterCmsSigner.SignedAttributes.Add(
+                     AttributeUtility.CreateNuGetV3ServiceIndexUrl(new Uri("https://api.nuget.org/v3/index.json")));
+                test.CounterCmsSigner.SignedAttributes.Add(
+                    AttributeUtility.CreateNuGetPackageOwners(new List<string> { "microsoft", "nuget" }));
+                test.CounterCmsSigner.SignedAttributes.Add(
+                    new Pkcs9SigningTime());
+                test.CounterCmsSigner.SignedAttributes.Add(
+                    AttributeUtility.CreateSigningCertificateV2(test.Certificate, HashAlgorithmName.SHA256));
+
+                // Create counter signature
+                test.PrimarySignedCms.SignerInfos[0].ComputeCounterSignature(test.CounterCmsSigner);
+
+                // Load primary signature
+                var primarySignature = PrimarySignature.Load(test.PrimarySignedCms.Encode());
+
+                // Validate countersignature
+                var countersignature = RepositoryCountersignature.GetRepositoryCountersignature(primarySignature);
+
+                Assert.Equal(SignatureType.Repository, countersignature.Type);
+            }
+        }
+
+        private sealed class LoadTest : IDisposable
```

Comment:
I think you copied this test helper class.  It was named `LoadTest` (elsewhere) because it was for testing the `Load(...)` method (e.g.:  `PrimarySignature.Load(...)`).  You can simply rename this to `Test` but don't keep it `LoadTest`.